agent/engine/docker_task_engine.go

```diff
@@ -299,6 +299,12 @@ func (engine *DockerTaskEngine) monitorExecAgentRunning(ctx context.Context,
 		seelog.Errorf("Task engine [%s]: Could not retrieve docker id for container", task.Arn)
 		return
 	}
+	// Sleeping here so that all the containers do not call inspect/start exec agent process
+	// at the same time.
+	// The max sleep is 50% of the monitor interval to allow enough buffer time
+	// to finish monitoring.
+	// This is inspired from containers streaming stats from Docker.
+	time.Sleep(retry.AddJitter(time.Nanosecond, defaultMonitorExecAgentsInterval/2))
```

Comment:
let's use `engine.monitorExecAgentsInterval` instead. I think using the default time will break the test monitor unit tests